build/Dockerfile

```diff
@@ -1,4 +1,14 @@
-FROM golang:1.13.7-alpine3.11 as builder
+FROM registry.access.redhat.com/devtools/go-toolset-rhel7:1.12.12-4 as builder
+
+ENV PATH=/opt/rh/go-toolset-1.12/root/usr/bin:$PATH \
+    GOPATH=/go/
+
+USER root
```

Comment:
Is using root required?

Reply:
> Do we need to start implementing/maintaining Dockerfile and rhel.Dockerfile?

If the plan is to build this in Brew, then yes. I still don't have a clear answer from @parag about what we're doing here.

Reply:
seems we are going the brew route and will be setting up repos via comet to host the code.

Reply:
for reference here's the latest CRW operator dockerfile. http://pkgs.devel.redhat.com/cgit/containers/codeready-workspaces-operator/tree/Dockerfile?h=crw-2.2-rhel-8 Note it's using go-toolset 1.12.

Reply:
Without user root I'm not even able to build to locally:
![Screenshot_20200601_133940](https://user-images.githubusercontent.com/5887312/83401439-70125080-a40d-11ea-805e-dc660e42b583.png)